pkg/agent/openflow/client.go

```diff
@@ -290,14 +292,22 @@ func (c *client) IsConnected() bool {
 
 // addFlows installs the flows on the OVS bridge and then add them into the flow cache. If the flow cache exists,
 // it will return immediately, otherwise it will use Bundle to add all flows, and then add them into the flow cache.
-// If it fails to add the flows with Bundle, it will return the error and no flow cache is created.
-func (c *client) addFlows(cache *flowCategoryCache, flowCacheKey string, flows []binding.Flow) error {
-	_, ok := cache.Load(flowCacheKey)
-	// If a flow cache entry already exists for the key, return immediately. Otherwise, add the flows to the switch
-	// and populate the cache with them.
-	if ok {
-		klog.V(2).Infof("Flows with cache key %s are already installed", flowCacheKey)
-		return nil
+// If it fails to add the flows with Bundle, it will return the error and no flow cache is created. If the force parameter
+// is true, flows will be added regardless of flow cache.
+func (c *client) addFlows(cache *flowCategoryCache, flowCacheKey string, flows []binding.Flow, force bool) error {
+	if !force {
+		_, ok := cache.Load(flowCacheKey)
+		// If a flow cache entry already exists for the key, return immediately. Otherwise, add the flows to the switch
+		// and populate the cache with them.
+		if ok {
+			klog.Infof("Flows with cache key %s are already installed", flowCacheKey)
+			return nil
+		}
+	} else {
+		err := c.ofEntryOperations.DeleteAll(flows)
```

Comment:
It's not ideal to remove previous flows first then install new ones as there could be a chance that existing traffic is affected. Can we do this in a bundle?
Also I feel the current implementation that `addFlows` only check whether the key matches but not the actions may cause problem in future. Ideally it should check whether the exact flows have been installed and skip if yet, otherwise it should remove the previous flows and install new ones with a bundle. To minimize the risk, maybe use a new method to implement the override behavior for Node flows only, other cases can switch to it later.

Reply:
Updated. Used a new method `modifyFlows()`.